src/runtime/policy/tests/policy-test.ts

```diff
@@ -352,10 +352,144 @@ policy MyPolicy {
     const capabilities = target.toCapabilities();
     assert.lengthOf(capabilities, 2);
     assert.isTrue(capabilities[0].isEquivalent(Capabilities.create([
-      Persistence.onDisk(), new Encryption(true), Ttl.days(2)
-    ])));
+        Persistence.onDisk(), new Encryption(true), Ttl.days(2)])),
+        `Unexpected capabilities: ${capabilities[0].toDebugString()}`);
     assert.isTrue(capabilities[1].isEquivalent(
-      Capabilities.create([Persistence.inMemory(), Ttl.days(2)])
-    ));
+        Capabilities.create([Persistence.inMemory(), new Encryption(false), Ttl.days(2)])),
+        `Unexpected capabilities: ${capabilities[1].toDebugString()}`);
+  });
+
+  const manifestString = (annotations = '') => {
+    return `
+${personSchema}
+particle P1
+  person: reads writes Person {name, age}
+recipe
+  personHandle: create ${annotations}
+  P1
+    person: personHandle
+    `;
+  };
+
+  it('fails validating handle with no policy target', async () => {
+    const policy = (await parsePolicy(`
+policy MyPolicy {
+  @allowedRetention(medium: 'Disk', encryption: true)
+  @maxAge('2d')
+  from Person access {}
+}
+    `));
+    const recipe = (await Manifest.parse(`
+particle P1
+  foo: reads writes Foo {value: Text}
+recipe
+  fooHandle: create @persistent @ttl('2d')
+  P1
+    foo: fooHandle
+    `)).recipes[0];
+    assert.isTrue(recipe.normalize() && recipe.isResolved());
+    assert.isFalse(policy.validateHandle(recipe.handles[0]));
+  });
+
+  const failHandleWithPolicy = async (manifestString, policy, expectedError) => {
+    const recipe = (await Manifest.parse(manifestString)).recipes[0];
+    assert.isTrue(recipe.normalize() && recipe.isResolved());
+    const options = {errors: new Map()};
+    assert.isFalse(policy.validateHandle(recipe.handles[0], options));
+    assert.isTrue([...options.errors.values()].join(' ').includes(expectedError),
+      `Expected "${[...options.errors.values()].join(' ')}"\n to contain: "${expectedError}"`);
+  };
+  const successfulValidation = async (manifestString, policy) => {
+    const recipe = (await Manifest.parse(manifestString)).recipes[0];
+    assert.isTrue(recipe.normalize() && recipe.isResolved());
+    const options = {errors: new Map()};
+    assert.isTrue(policy.validateHandle(recipe.handles[0], options),
+        `Validation failed with: ${[...options.errors.values()].join(' ')}`);
```

Comment:
Again, I can't help but think this pattern where you pass in an errors map as a param is a bit unergonomic. 

Just spitballing here: [Playground Link](https://www.typescriptlang.org/play/#code/MYGwhgzhAEBqYgJYBMwBdEHsB2AlAphAK4hrQA8AKgHzQDeAsAFDSvT4BOHmHEAXNACyYAA5UANNAhoOibAHNaAXmjZ8AdyGiAFAEoA3M2Zto8-GWLBghCHvrQO5oh2zQ0AC0QQAdJ269vCEQAL3xoJQjoAAZ9aABfIxY2MGRkAFEuHm0Aa3wATwFKST8eAWlZBV16YxM2Dy9fTICIcxz84qaDGtYEpm7oFOQCYlJtTA9OAXgkVAwcYZI0KmoqxiTa6HH3Tkb-HwAzHjSwYHdtbQA3SWyqpVp6nxLm1uzJC91dft7+tEwAZRkcnkdjWG1YiH20G0D0CRCsNlW-TBDicLmgAHJpih0FhXI4RmgBH84dYoOikWw4uwQC1qutkSi0M5XJiENi5njCIsBAAxMCIEDOfAAHWw6OgAGpoABtbxymFPHwXBBEQh6AC63gAVpg5Np0aL0V16SZeqbmN8mGg8iIwgAFTBIYB5cL0OKGPpMfZEbDADnQZUzdD4P6YAC25k8Cj0UzZs1xC1I5AdTrytFBbGAOGkKIJrrUmix8fmXKTKcQzuoeg9Jnxi28gwy-m0dDikgABk2eNAI1AwGZoNtHO3jbXUZyCR7LUA)

Usage in this test would then be:

```
const result = policy.validateHandle(...);
assert.isTrue(result.success, `Validation failed with: ${result}.`);
```

Reply:
updated the code (sorry, i was a bit confused why the playground had 2 options - were these options to choose from? i went with the one on the right) either way, i'm happy to address further comments in follow up prs

Reply:
Left is TS, right is the compiled JS output